### ISSUE TITLE
Sprint 17.01.25 - Committed

### ISSUE THREAD

**Issue by williamspatrick** [openbmc/openbmc#976]
-------------------------------------------------------------------------------------------------
**Andrew:geissonator -** 

**17.01.25:**
- https://github.com/openbmc/openbmc/issues/1033 (handle service failures)
    - 01/26: Email started on distro, discussed first steps in design meeting
    - 01/30: Good discussion with Michael, Patrick, Brad, got initial quiesce code up
    - 01/31: Experimented with OnFailure= tag, have plan to discuss with Brad
    - 02/02: Sent out email to distro, did more testing, need to find solution to running same target twice.
    - 02/03: Did some more prototyping, have final solution proposal, will start coding
    - 02/06: Documentation up https://gerrit.openbmc-project.xyz/#/c/2253/ - will work on target updates today.  May need to do a quick fix for P9 to get HOST_BOOTING back.
    - 02/07: Got the targets and service files updated and in gerrit, testing found some issues, refactor coming today.
-------------------------------------------------------------------------------------------------
**Brad:bradbishop -**
**17.01.25:**
- https://github.com/openbmc/openbmc/issues/956 (add object lifecycle yaml for inventory from ipmi)
    - 01/30: Worked all weekend on this, looking good, will get task list up.
    - 02/01: 8 more commits last night, all comments addressed
    - 02/02: Deepak looked, Andrew to review
    - 02/03: Added phosphor dbus interfaces to native sdk 
    - 02/06: Code up, been working the reqs.  Worked Patrick set properties code on Friday, has it working now, working on inventory changes to use it.  Brad to get an openbmc build repo on github that has everything.
    - 02/07: Needs reviews and merges
- https://github.com/openbmc/openbmc/issues/995 (create error if BMC EEPROM not found)
    - 01/26: Will create task lists
Inventory
    - 02/03: Worked on Patrick's new dbus code.  Goal today is have a build with inventory manager that makes properties
    - 02/07: Spent yesterday getting a fork together with all the required inventory features.
-------------------------------------------------------------------------------------------------
**Adriana:anoo1 -**
**17.01.11:**
- https://github.com/openbmc/openbmc/issues/352 (hostboot eSEL to openbmc elog)
    - 01/26: Above one is pre-req, then have some comments here
    - 01/30: All code up there, rebased, ready for re-review Vishwa/Deepak/Andrew/Patrick.  
    - 01/31: Will address Deepak comments today
    - 02/02: Patrick reviewed and approved, but need to wait for CI
    - 02/03: Worked with Brad to get recipes merged, 7 commits ready now, just nee to be re-run through CI.  
    - 02/06: Brad to merge commits that passed CI on Friday, have some remaining code review comments to address.
    - 02/07: Last eSEL changes were pushed up into gerrit.  Andrew and Deepak to review.  Push Adriana's code into Brad's fork so we have a repo for mfg stage 2 testing.

**17.01.25:**
- https://github.com/openbmc/openbmc/issues/1035 (UBI/UBIFS support in kernel)
    - 01/30: Figured out kernel config file option, will do today.
    - 01/31: Built, need to try it out.
- https://github.com/openbmc/openbmc/issues/1036 (Verify UBI volumes can be created)
-------------------------------------------------------------------------------------------------
**Michael:mtritz -**
**17.01.25:**
- https://github.com/openbmc/openbmc/issues/942 (disable automatic reboots of host)
    - 01/26: Need task break down
    - 01/27: Tasks are up, need some tech discussion
    - 01/30: Good conversation on Friday, will start working today
    - 01/31: Initial settingsd change is up, will stop by and talk with Andrew today
    - 02/02: Will stop in to see Andrew
    - 02/03: Initial host checkstop code is up for looking a
    - 02/06: Working on testing the checkstop code, ran into issues with testing this week.
    - 02/07: Got the checkstop code tested, working on the state management part now
-------------------------------------------------------------------------------------------------
**Gunnar:gtmills**
**17.01.11:**
- https://github.com/openbmc/openbmc/issues/670 (fw version implementation)
    - https://gerrit.openbmc-project.xyz/#/c/1879/
    - 01/26: Uploaded new patches, Patrick pinged about active interface, Gunnar adding to it.  Andrew and Vishwa
    - 01/27: Need Vishwa and/or Josh reviews, maybe go right to Patrick
    - 01/30: Worked on active interfaces, has some code feedback, needs re-review.  Vishwa or Lei to look next.
    - 01/31: Code reviews from Josh needed.  Recipe is up, will add Brad.
    - 02/01: Working on Patrick code review comments
    - 02/02: Will push up code for review comments today
    - 02/03: Pushed a bunch of new commits to address Patrick comments, Andrew to re-review
    - 02/06: Review comments over weekend, working on them
    - 02/07: Mostly worked on other work, new code up to look at for code reviews

**17.01.25:**
- https://github.com/openbmc/openbmc/issues/1037
    - 01/27: We need to scope this, break into 2 parts.  Write a tool that takes pnor image, makes squashfs, metadata descriptor file for TOC, then what about read/write sideless.
    - 01/30: Will start working today.  Adriana to rename and open 2 more based on Friday disucussion.
    - 01/31: Started looking at PNOR image and squashfs
    - 02/01: Patrick created repo, some comments from LTC team to look at
    - 02/03: Hope to get something up today
    - 02/06: First commits up
-------------------------------------------------------------------------------------------------
**Matt:spinler -** 
**17.01.11:**
- https://github.com/openbmc/openbmc/issues/629 (integrate device tree into kernel)
    - 01/26: Matt to follow up
    - 01/27: Did a lot of yocto research, seems like current approach is what they recommend but it does make the image "dirty".  Needs help.  Recommendation is new story to do out of tree, put what we got in for now.  So get this story in, then follow up with another story.  Patrick to talk with Joel.
    - 01/30: Made some changes in kernel, need to update scripts to handle this now
    - 01/31: Almost done
    - 02/01: Put up 11 commits yesterday for this, move to next sprint
    - 02/02: Need Andrew/Gunnar to review these commits
    - 02/03: Need Andrew/Gunnar to review these commits
    - 02/03: Gunnar reviewed, Andrew to review.  

**17.01.25:**
- Hostboot IPMI bringup on Witherspoon/Zaius (Patrick has issue on hostboot team to enable)
    - 01/26: Matt to talk with Jamie
    - 01/27: Needs date from Bill
    - 01/30: Norm started this bringup on Friday, have issues with eSEL's since
    - 02/07: Did some Zaius updates so we'll update things correctly with host ipmi enabled
- https://github.com/openbmc/openbmc/issues/1003 (add witherspoon FRU id’s to MRW)
    - 01/26: Waiting on Sertac to fix MRW
    - 01/27: Sertac working on it
    - 01/30: Need input from Sertac
    - 01/31: Need input from Sertac
    - 02/01: Need input from Sertac
    - 02/02: Need input from Sertac
    - 02/03: Got some MRW data from Sertac at midnight last night, Matt may fix
    - 02/06: Fixed up MRW, need IPMI sensors from Sertac, he's on vacation after today.
    - 02/07: Sertac did not provide sensor update, Matt may need to do it.
- https://github.com/openbmc/openbmc/issues/781 (functional IPMI state to inventory mapping) - depends on Brad
    - 01/31: Need to start looking into this one
    - 02/01: Still need to start looking into this one
    - 02/02: Talked with Brad, need to come up with some yaml to map inventory to sensors.  Right now it's all hard code in data files.
    - 02/03: Looked into this, knows pretty much what to do, need to handle the legacy sensor settings (like boot count from hostboot)
-------------------------------------------------------------------------------------------------
**Ratan:ratagupt -**
**17.01.25:**
- https://github.com/openbmc/openbmc/issues/983 (SLP logging too much to journal)
- https://github.com/openbmc/openbmc/issues/985 (Avahi fails to start on Zaius)
- https://github.com/openbmc/openbmc/issues/1022 (ipmi - call inventory manager Notify())
    - 01/30: Started working this today
- https://github.com/openbmc/openbmc/issues/1021 (map custom IPMI properties)
    - 01/30: This will only be CPU, shouldn't be too bad

- Worked barreleye and palmetto issue (need issue number)
-------------------------------------------------------------------------------------------------
**Tom:tomjoseph83 -**

**17.01.11:**
- https://github.com/openbmc/openbmc/issues/870 (encrypted rmcp, part 2)
    - 01/27: Code is up, need Deepak/Matt B. reviews
    - 01/30: Matt reviewed, Tom to address
    - 01/31: More code review comments
    - 02/01: Some code review comments to address
    - 02/02: Waiting on Patrick review

**17.01.25:**
- https://github.com/openbmc/openbmc/issues/996 (IPMI Add SEL fails)
    - 02/03: Will take a look next
    - 02/07: Not seen with later drivers, still an issue?  Tom to verify new event log created.
- https://github.com/openbmc/openbmc/issues/856 (Attach RMCP session to SOL, part 1)
    - 01/27: Tasks broken down
    - 01/30: Will discuss design tomorrow
    - 02/02: Started working on this today
    - 02/03: Started working the code, no reviews yet
    - 02/06: Interfaces done, will be pushing up tomorrow
    - 02/07: Interface file is up, needs a few more patches then add reviewers
-------------------------------------------------------------------------------------------------
**Deepak:** 
**17.01.25:**
- Overall mfg stage 2 interface to tools team
    - 01/30: Looked into final pieces for inventory, doing code reviews
    - 01/31: Lots of code reviews today
    - 02/01: Inventory tasks, got open power vpd tool working, updated mac address's
    - 02/02: Started doing inventory work on our new witherspoon
    - 02/07: Pulled Brad's latest github fork, was able to create inventory objects with valid properties on witherspoon box. 
- https://github.com/openbmc/openbmc/issues/1028 (server process callout design)
- https://github.com/openbmc/openbmc/issues/1023 (openpower - call inventory manager notify())
    - 01/30: Need Uma to update our VPD?  Yes, Matt S. to ask.
    - 02/02: Got this working today for FRU vpd.  Need Brad code and #970 from Patrick.
    - 02/03: Worked on this today and #1022 
    - 02/06: Pulled latest from inventory manager, tested out new MRW from Matt, verified new code, code up for review.
    - 02/07: Code up

-------------------------------------------------------------------------------------------------
**Vishwa:vishwabmc -**
**17.01.11:**
- https://github.com/openbmc/openbmc/issues/802 (move all existing code to new LED interfaces)
    - 02/02: Identified required changes and updated github issue.
    - 02/06: Started making the changes
    - 02/07: Today had to work on tech approval doc
- https://github.com/openbmc/openbmc/issues/930 (soft power off in host ipmi with watchdog timers)
    - https://gerrit.openbmc-project.xyz/#/c/1972/
    - 01/27: Code reviews going back and forth, back to Vishwa
    - 02/01: All patches up for review now, need to address new sdbusplus feature from Patrick
    - 02/02: Need code review
    - 02/03: Code review comments addressed
    - 02/06: Started working on the gtests, needs another round of code review
    - 02/07: Need re-review

**17.01.25:**
- Vacation, focus on code reviews and above stories
-------------------------------------------------------------------------------------------------
**Eddie:eddiejames -**
**16.12.28:**
- https://github.com/openbmc/openbmc/issues/624 (initial occ intf refactor)
- https://github.com/openbmc/openbmc/issues/625 (p9 occ data)
    - 01/26: Kernel feedback was make it work for either endian, tough though, asked them if he can just do our current way now.
    - 01/27: Sent new patch set up, not doing the endian question for now
    - 01/30: Gunter sent back response, Eddie to work today
    - 01/31: Fixed everything, then ran into gmail issue mailing out patches, moving over to LTC mailing list
    - 02/01: Send another patch set yesterday morning with new email, haven't heard back yet.
    - 02/02: Still waiting for feedback
    - 02/03: Will check for feedback
    - 02/06: Will check for feedback
    - 02/07: Sent out patch set yesterday

**17.01.11:**
- FSI based I2C driver (RTC story?)
    - 01/27: Worked on optimizing the i2c speeds
    - 02/01: Fixed more i2c bugs that Matt found in testing
    - 02/02: Talked with Joel about i2c stuff, will get base framework
    - 02/06: Will look for Joel i2c base today, working with Matt L. on testing
    - 02/07: Fixing issues that Joel brought up in testing

**17.01.25:**
- Continue i2c bringup and upstream efforts?
    - SCOM reset function
    - Working on this some more and testing today
    - Finished up this
    - Looking into SBE driver
    - 02/03: Build final image, make sure cronus still is good
-------------------------------------------------------------------------------------------------
**Chris -**
**17.01.25:**
- https://github.com/openbmc/openbmc/issues/1038 (pdbg work for mfg stage 2 story)
    - 01/27: Mostly coded up now, running tests.  Dug into pdbg code, quite a bit will need to be disabled.  Will need to talk with Alistair.  Get code up in gerrit.  Patrick to create repo in gerrit.
    - 01/30: Continued pdbg user interface work, running some tests.  Joel commented that we need to be sure to discuss with Alistair.  Chris to document and send email.  Patrick to create repo.  Will discuss today.
   - 02/01: Per Monday call, put this on hold and use sysfs.

- openFSI
    - 02/01: Get code up to openbmc for Joel to take in
    - 02/02: Sent out version 3 to community for review, no feedback yet.  Put up interrupt code to openbmc mailing list (required for i2c code).  Today a few more pieces for nack.   Some risk in plan to try and get i2c code in.
    - 02/03: Got most of required code merged into openbmc kernel last night, Chris to verify it's in and working as expected.  Need to enable FSI and I2C configs in kernel once we get pdbg plan in place (sysfs interface).
    - 02/06: Submitted some more patches to Joel, Joel opened some issues and made some patches.  Chris to look at those today.
    - 02/07: Worked Joel openFSI issue, #1086 Has fix, needs testing, and then will get patch out.  #1085 is next.
-------------------------------------------------------------------------------------------------
**Josh:JoshDKing -** 
**16.12.28**
- https://github.com/openbmc/openbmc/issues/772 (BMC state mgmt and control)
    - Final push up coming from Patrick review feedback
    - 01/27: Code merged, working on final piece.
    - 01/31: Will do a check on startup for state
    - 02/02: Think he's found solution to getting startup state
    - 02/03: More coding yesterday
    - 02/07: Talked with George about reboot issue, looking into.  
   
**17.01.11:**
- State screen/bringup
- obmcutil updates for state

**17.01.25:**
- https://github.com/openbmc/openbmc/issues/938 (log.hpp vs phosphor-logging/log.hpp include)
-------------------------------------------------------------------------------------------------
**Matt B.:msbarth -** 
**17.01.11:**
- https://github.com/openbmc/openbmc/issues/959 (fan presence detection)
    - 01/26: Getting daily feedback from Brad, working on config file
    - 01/27: More discussions with Brad/Matt to figure out yaml file, working on parser today
    - 01/30: Got python and yaml code up, started working on getting makefile and such into build.
    - 01/31: Ran into integration issues in build process, think he has solution
    - 02/01: Got new commit up that runs fan script wedge, back to other commit comments
    - 02/02: Addressed all comments, will start using inventory items soon
    - 02/03: More discussions with Brad yesterday, brainstorming inventory usage, about 1/3 done
    - 02/06: Got all the code reorganized on Friday, will push up commits today.  Still need inventory and callback code.
    - 02/07: Fixed up all the commits, put up a new one for dbus signal framework with callback.  Started working on inventory stuff based off of Deepak's stuff.

**17.01.25:**
- TBD
-------------------------------------------------------------------------------------------------
**Nan:williamli80 -**
**17.01.11:**
- https://github.com/openbmc/openbmc/issues/770 (pciedetect refactor)

**17.01.25:**
- Vacation, carry-over
-------------------------------------------------------------------------------------------------
**Yi:adamliyi -**
**16.12.28:**
- https://github.com/openbmc/openbmc/issues/767 (host watchdog refactor)
    - https://gerrit.openbmc-project.xyz/#/c/1513/

**17.01.11:**
- https://github.com/openbmc/openbmc/issues/665 (FRU vpd in inventory)

**17.01.25:**
- Vacation, carry-over
-------------------------------------------------------------------------------------------------
**Lei:mine260309 -** 
**16.12.28:**

**17.01.11:**
- https://github.com/openbmc/openbmc/issues/964 (romulus rtc driver)

**17.01.25:**
- Assist Nan/Yi
-------------------------------------------------------------------------------------------------
**Patrick:williamspatrick -**
**16.12.28**
- https://github.com/openbmc/openbmc/issues/477 (format to document org.openbmc interfaces)
    - 01/27: Non critical
    
**17.01.11:**
- https://github.com/openbmc/openbmc/issues/851 (client proxy bindings) and https://github.com/openbmc/openbmc/issues/970
    - 01/27: Will work on this today, discussed with Brad
    - 01/30: Spent Friday getting library built for phophor-dbus-interfaces and linking against other repos (needed by Adriana)
    - 01/31: Code up for client libraries, need tool from Matt to get CI passing
    - 02/01: Patrick to get this critical task up this afternoon

**17.01.25:**
- Vacation
